src/modules/userSyncTest/downloadSize.js

```diff
@@ -0,0 +1,13 @@
+import { memorySizeOf } from '../../utils/memory-size'
+
+export const getDownloadSize = dataArray => {
+    let downloadSize = 0
+    dataArray.forEach(dataElement => {
+        const dataSize = parseFloat(
+            memorySizeOf(JSON.stringify(dataElement.toArray()))
```

Comment:
`JSON.stringify` is very slow.  This can be a TODO issue for later, but we should look into how it performs here.